src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,43 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 4;
+    private int[] array;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+    }
+
+    @Override
+    protected void compute() {
+        if (array.length <= THRESHOLD) {
```

Comment:
The threshold value of 4 is quite low for using `Arrays.sort`. Consider increasing it to a higher value, such as 1000, to better utilize the benefits of parallel processing for larger arrays.